samples/bluetooth/bthome_sensor_template/src/main.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/uuid.h>
+
+#define SERVICE_DATA_LEN        9
+#define SERVICE_UUID            0xfcd2		/* BTHome service UUID */
+
+#define ADV_PARAM BT_LE_ADV_PARAM(BT_LE_ADV_OPT_USE_IDENTITY, \
+					     BT_GAP_ADV_SLOW_INT_MIN, \
+					     BT_GAP_ADV_SLOW_INT_MAX, NULL)
+
+static uint8_t service_data[SERVICE_DATA_LEN] = {
+	BT_UUID_16_ENCODE(SERVICE_UUID),
+	0x40,
+	0x02,	/* Temperature */
```

Comment:
could you define a packed struct instead? I think it'd help developers learning from this sample.

Reply:
Defining a struct for temperature and humidity is possible but then this will not be a generic template anymore since there are many more sensor types you can have. See supported datatypes here:
[Supported data types](https://bthome.io/format/)

Reply:
Agreed. Using a packed struct here would make it less a template.

Reply:
you can still have a struct for the two object types, and include that struct into the service_data struct. But you're not implementing a library, just a sample so that's ok.